### PR TITLE
GEOPY-741- Airborne FEM survey

### PR DESCRIPTION
**Pull request by benk-mira**
**GEOPY-741 - Airborne FEM survey**

### CODE REVIEW

File: geoh5py/objects/surveys/electromagnetics/base.py [hunk 1]
```diff
@@ -193,10 +194,70 @@ def components(self) -> dict | None:
...
+
+        metadata["EM Dataset"][new_entity.type] = new_entity.uid
+
+        if apply_to_complement and (getattr(self, "complement", None) is not None):
+            base_object = (
+                self.base_transmitter_type
+                if isinstance(self, self.default_receiver_type)
+                else self.base_receiver_type
+            )
+            new_complement = super(base_object, self.complement).copy(  # type: ignore
```
Comment (domfournier) [inline geoh5py/objects/surveys/electromagnetics/base.py:239]: Damn, that's nice.

File: geoh5py/objects/surveys/electromagnetics/ground_tem.py [hunk 2]
```diff
@@ -83,40 +84,41 @@ def copy(
...
                 ]
-                mask = np.zeros(complement.vertices.shape[0], dtype=bool)
-                mask[complement.cells[cell_mask, :]] = True
-                tx_ids = complement.tx_id_property.values[cell_mask]
+                mask = np.zeros(self.complement.vertices.shape[0], dtype=bool)
+                mask[self.complement.cells[cell_mask, :]] = True
+                tx_ids = self.complement.tx_id_property.values[cell_mask]
 
-            new_complement = super(Curve, complement).copy(  # type: ignore
+            new_complement = super(Curve, self.complement).copy(  # type: ignore
```
Comment (domfournier) [inline geoh5py/objects/surveys/electromagnetics/ground_tem.py:121]:
```suggestion
            new_complement = super(base_object, self.complement).copy(  # type: ignore
```

File: geoh5py/objects/surveys/electromagnetics/base.py (hunk 1 above)
Comment (domfournier) [reply geoh5py/objects/surveys/electromagnetics/base.py:239]: Maybe `complement` should be default to None on the base class?

**Review by domfournier — APPROVED** [review]
Very nice. Minor change below that could help with pylint issues

File: geoh5py/objects/surveys/electromagnetics/ground_tem.py (hunk 2 above)
Comment (benk-mira) [reply geoh5py/objects/surveys/electromagnetics/ground_tem.py]: Done

File: geoh5py/objects/surveys/electromagnetics/base.py (hunk 1 above)
Comment (benk-mira) [reply geoh5py/objects/surveys/electromagnetics/base.py]: Done

File: geoh5py/objects/surveys/electromagnetics/airborne_fem.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+from geoh5py.objects.curve import Curve
+from geoh5py.objects.object_base import ObjectType
+
+from .base import AirborneEMSurvey, FEMSurvey
+
+
+class AirborneFEMSurvey(
+    FEMSurvey, AirborneEMSurvey
+):  # pylint: disable=too-many-ancestors
```
[the remark this answers is not in the document]
Comment (benk-mira) [reply geoh5py/objects/surveys/electromagnetics/airborne_fem.py:30]: I ended up putting them at the top of the affected files

**Review by domfournier — DISMISSED** [review]
Looks great. Thanks for this.

File: geoh5py/objects/surveys/electromagnetics/airborne_fem.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (domfournier) [reply geoh5py/objects/surveys/electromagnetics/airborne_fem.py:30]: Looks like you might have forgotten that one?

**Review by domfournier — APPROVED** [review]
Wil need to update docs on GEOPY-923

**Pull request merged**